scripts/build-circt.sh

```diff
@@ -43,12 +48,15 @@ while [[ "$#" -ge 1 ]] ; do
 	esac
 done
 
-git clone https://github.com/EECS-NTNU/circt.git ${CIRCT_BUILD}
-cd ${CIRCT_BUILD}
+if [ ! -d "$CIRCT_BUILD/circt.git" ] ;
+then
+	git clone https://github.com/EECS-NTNU/circt.git ${CIRCT_BUILD}/circt.git
+fi
+cd ${CIRCT_BUILD}/circt.git
 git checkout ${GIT_COMMIT}
-mkdir build
-cd build
-cmake -G Ninja .. \
+mkdir -p  ../build
```

Comment:
A separate build directory is used for building outside the source directory. 
Same as jlm is build in build-release/build-debug.
I've opted for the following directory structure
```
CIRCT_BUILD
    build     # CIRCT_BUILD_DIR
    circt.git # CIRCT_GIT_DIR
```

An option would be to clone into CIRCT_BUILD and then only have a build subdirectory, CIRCT_BUILD/build.

I find the current setup to me more clear, since a user would expect to find the build in CIRCT_BUILD, but we also need to place the source somewhere. So now when somebody looks into the build directory, it should be rather obvious what is what.